compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BFiniteType.java

```diff
@@ -65,6 +65,41 @@ public String toString() {
         return joiner.toString();
     }
 
+    @Override
+    public boolean hasImplicitInitialValue() {
+        // Has NIL element as a member.
+        if (valueSpace.stream().anyMatch(value -> value.type.tag == TypeTags.NIL)) {
+            return true;
+        }
+        // All of the element are from same type. And elements contains implicit initial value.
+        if (!valueSpace.isEmpty()) {
+            BLangExpression firstElement = valueSpace.iterator().next();
+            boolean sameType = valueSpace.stream().allMatch(value -> value.type.tag == firstElement.type.tag);
+            if (!sameType) {
+                return false;
+            }
+
+            switch (firstElement.type.tag) {
+                case TypeTags.STRING:
```

Comment:
What will happen if the type is decimal?

Reply:
I don't think we can define finite type for decimals as we don't have a decimal literal.